### PR TITLE
Allow for use of meta-query

### LINKED ISSUE #86: Scan for all the mDNS devices in the network
**Is your feature request related to a problem? Please describe.**
I would like to scan all mDNS devices in the network in an efficient manner.
I saw that there is an option to search for a specific serviceType but I prefer not to search all the serviceTypes that ever existed.

**Describe the solution you'd like**
As written in [DNS-SD specification](https://datatracker.ietf.org/doc/html/rfc6763#section-9) chapter 9 there is a special type `_services._dns-sd._udp` that retrieves all current types in the network.

I would love to see a function that retrieves all of the serviceType on the network using _services._dns-sd._udp so that I can iterate only on existing services in the network.

**Describe alternatives you've considered**
I have searched also in [multicast_dns](https://pub.dev/packages/multicast_dns) but they do not support it yet, see issue number 97210 on flutter repo https://github.com/flutter/flutter/issues

Also [nsd](https://pub.dev/packages/nsd) package had implemented it in issue number 8 https://github.com/sebastianhaberey/nsd/issues/ .
This package is more popular, supports Linux, I think the other package does not support iOS (issue  57) so it would be easier for me if it exists here.

### PR DESCRIPTION
**Pull request by Caesarovich**
Hi,

As discussed in #86, it is normally possible to use the meta-query `_services._dns-sd._udp` to scan for every available services on the network. Enabling to then scan/discover every services and devices on the network. But it is currently not possible due to two things:

1. The `normalizeType` function that doesn't allow the specific query.
2. The platform specific implementations that are unable to deal with this specific case.

This PR fixes the first problem and also the platform implementation for Linux. I'm currently working on getting it to work for Android. But I won't be able to do it for Apple devices. 

I'm waiting to hear back from you regarding this PR.

-> Fixes #86

### CODE REVIEW

**Comment by Caesarovich** [comment]
Hi @Skyost, I noticed there hasn't been a review given to this PR yet.

It's okay as I was away from my current projects for a bit, but could you please have a look at it ?

Thanks !

**Comment by Skyost** [comment]
Hey,    
I thought it was unfinished as you said "This PR fixes the first problem and also the platform implementation for Linux. I'm currently working on getting it to work for Android.".

**Comment by Caesarovich** [comment]
Hi @Skyost, I did not realize I forgot to do an update on this. So I went back in the same "rabbit hole" as last time I was working on this.

TLDR is,
The code in Bonsoir works perfectly good for this case on Android so there's no need to change it. The issue lies in Android itself, the meta-query stops working at the release of Android 13 (API 33). I've tested it with emulators and everything is fine before that major version. I've been trying to pin-point where the regression happened but the Android codebase is way too large for me to be able to do anything. 

Also I didn't mention it in the original post but it works on Windows as well.

So for the scope of this PR there is nothing else to add.

For the Android issue, do you want to open an issue upstream ? I'll do it otherwise.

**Comment by Skyost** [comment]
That's okay, thanks a lot !
